jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RewriteEncodingRule.java

```diff
@@ -0,0 +1,285 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.rewrite.handler;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.http.PreEncodedHttpField;
+import org.eclipse.jetty.server.Context;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.TypeUtil;
+import org.eclipse.jetty.util.URIUtil;
+import org.eclipse.jetty.util.resource.Resource;
+
+/**
+ * Rule that can rewrite a request based on the {@code Accept-Encoding} header.
+ */
+public class RewriteEncodingRule extends Rule
+{
+    public static Resource getResource(Context context, String pathInContext)
+    {
+        return context.getBaseResource().resolve(pathInContext);
+    }
+
+    protected record Encoding(String encoding, String extension, HttpField contentEncodingField)
+    {
+        public static Encoding of(String encoding, String extension)
+        {
+           return new Encoding(encoding, extension, new PreEncodedHttpField(HttpHeader.CONTENT_ENCODING, encoding));
+        }
+    }
+
+    private static final HttpField VARY_ACCEPT_ENCODING = new PreEncodedHttpField(HttpHeader.VARY, HttpHeader.ACCEPT_ENCODING.asString());
+    private final BiFunction<Context, String, Resource> _getResource;
+    private final Map<String, Encoding> _encodings = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+
+    protected RewriteEncodingRule()
```

Comment:
All constructors are `protected`, so it won't be possible to use this rule in XML.
I think they should be `public`.